node/src/components/consensus/era_supervisor.rs

```diff
@@ -408,9 +425,15 @@ where
         let scale_stake = |(key, stake): (PublicKey, Motes)| {
             (key, AsPrimitive::<u64>::as_(stake.value() / scaling_factor))
         };
-        let validators: Validators<PublicKey> =
+        let mut validators: Validators<PublicKey> =
             validator_stakes.into_iter().map(scale_stake).collect();
 
+        for e_id in era_id.iter_other_bonded() {
+            for pub_key in &self.active_eras[&e_id].banned {
+                validators.ban(pub_key);
+            }
+        }
```

Comment:
Shouldn't those validators be part of `banned` as well?

Reply:
Once [NDRS-495](https://casperlabs.atlassian.net/browse/NDRS-495) is done, `banned` will only contain the validators that equivocated in the previous era, or at the end of the one before that. So we'll need to collect all from the last `BONDED_ERAS` switch blocks.